### PR TITLE
feat(std/testing): Add support for object assertion against object subset 

### PR DESCRIPTION
**Pull request by lowlighter**
This add supports for a new assertion function `assertObjectMatch` which allows to test an `actual` object against an `expected` object subset (i.e. inclusivity, not equality)

**Why ?**

It's often useful to test an object partly without testing the remaining properties (like for web api response, where you might want to only check a few fields).

This function filters out `actual` object with `expected` object subsets keys before calling `assertEquals` (so it's actually a wrapper which acts as a syntaxic sugar).

```ts
const a  = {
   foo: true,
   bar: true,
   baz: {
     qux: true
   }
}

// Simple assertions
assertObjectMatch(a, {foo: true})  // OK 
assertObjectMatch(a, {bar: true})  // OK
assertObjectMatch(a, {foo: false}) // KO 

// Nested assertions
assertObjectMatch(a, {baz: {qux: true}}) // OK
assertObjectMatch(a, {baz: {qux:null}})  // KO
```

### CODE REVIEW

**Comment by bartlomieju** [comment]
Thanks for the contribution @lowlighter! Is there some prior art for this assertion? 

@nayeemrmn PTAL

File: std/testing/asserts.ts
Comment (nayeemrmn) [inline std/testing/asserts.ts:438]: Should symbols be allowed?

**Comment by lowlighter** [comment]
> Is there some prior art for this assertion?

Yes, it's heavily inspired from [expect.toMatchObject(object)](https://jestjs.io/docs/en/expect.html#tomatchobjectobject) from [Jest](https://jestjs.io/en) JavaScript testing framework, though implementation may differ a bit. Since a lot of logic has already been done in `assertEquals`, I thought it would easier to filter the actual object with the expected one so it'll behave the same.


> Should symbols be allowed?

Yes, I'll add them and add tests to cover them !

**Comment by bartlomieju** [comment]
> Yes, it's heavily inspired from expect.toMatchObject(object) from Jest JavaScript testing framework, though implementation may differ a bit. Since a lot of logic has already been done in assertEquals, I thought it would easier to filter the actual object with the expected one so it'll behave the same.

@lowlighter thanks, that sounds good to me. I'll wait for @nayeemrmn's review before landing

File: std/testing/asserts.ts
Comment (nayeemrmn) [inline std/testing/asserts.ts:453]: I don't think `Object.entries()` includes symbols. This would be shown if the tests included an erroneous case for symbols. You can use `[...Object.getOwnPropertyNames(a), ...Object.getOwnPropertySymbols(a)]` to get a complete list of keys.

Comment (lowlighter) [reply std/testing/asserts.ts]: I replaced `Object.entries()` with your suggestion and it works like a charm, and changed `string|symbol` by `PropertyKey` which seems more accurate.

I got some trouble with using `symbol` as index signature so I casted them into `string` but not sure it's the cleanest way...

I wanted to add an erroneous case like you suggested, but seems that `assertEquals` does not handle `symbol`s perfectly currently (unless I missed something) : 
```ts
import { assertEquals } from "https://deno.land/std/testing/asserts.ts"
const foo = Symbol("foo")
Deno.test("symbol1", () => assertEquals({[foo]:"bar"}, {[foo]:"bar"}))
Deno.test("symbol2", () => assertEquals({[foo]:"bar"}, {[Symbol("foo")]:"bar"}))
```
```
running 2 tests
test symbol1 ... ok (2ms)
test symbol2 ... ok (1ms)

test result: ok. 2 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out (4ms)
```

**Review by nayeemrmn — APPROVED** [review]
LGTM

**Review by bartlomieju — APPROVED** [review]
LGTM, thanks @lowlighter

**Pull request merged**
